host/src/c_api/c_api.cc

```diff
@@ -846,13 +857,33 @@ XGB_DLL int XGDMatrixGetUIntInfo(const DMatrixHandle handle,
 }
 
 XGB_DLL int XGDMatrixNumRow(const DMatrixHandle handle,
-                            xgboost::bst_ulong *out) {
-  safe_ecall(enclave_XGDMatrixNumRow(Enclave::getInstance().getEnclave(), &Enclave::getInstance().enclave_ret, handle, out));
+                            uint8_t* nonce,
+                            size_t nonce_size,
+                            uint32_t nonce_ctr,
+                            xgboost::bst_ulong *out,
+                            char **signers,
+                            uint8_t* signatures[],
+                            size_t* sig_lengths) {
+  size_t signer_lengths[NUM_CLIENTS];
+  for (int i = 0; i < NUM_CLIENTS; i++) {
```

Comment:
Looks like this block of code
```
for (int i = 0; i < NUM_CLIENTS; i++) {
    signer_lengths[i] = strlen(signers[i]);
  }
```

was added to many functions. Let's create a helper function called `get_signer_lengths` (or something) for it?